scipy/stats/_multivariate.py

```diff
@@ -3521,3 +3522,353 @@ def rvs(self, eigs, random_state=None, tol=1e-13, diag_tol=1e-7):
         return m
 
 random_correlation = random_correlation_gen()
+
+
+_nig_doc_default_callparams = """\
+loc : scalar, optional
+    Mean of the distribution (default zero)
+variance_scale : positive scalar, optional
+    Scale on normal distribution prior (default one)
+shape : positive scalar, optional
+    Shape of the distribution (default one)
+scale : positive scalar, optional
+    Scale on inverse gamma distribution prior (default one)
+"""
+
+_nig_doc_callparams_note = \
+    """The parameters `loc` and `variance_scale` serves as the
+    hyperparameters for normal distribution which models the
+    unknown mean. The parameters `shape` and `scale` serves as
+    the hyperparameters for inverse gamma distribution which
+    models the unknown variance.
+    """
+
+_nig_doc_frozen_callparams = ""
+
+_nig_doc_frozen_callparams_note = \
+    """See class definition for a detailed description of parameters."""
+
+nig_docdict_params = {
+    '_nig_doc_default_callparams': _nig_doc_default_callparams,
+    '_nig_doc_callparams_note': _nig_doc_callparams_note,
+    '_doc_random_state': _doc_random_state
+}
+
+nig_docdict_noparams = {
+    '_nig_doc_default_callparams': _nig_doc_frozen_callparams,
+    '_nig_doc_callparams_note': _nig_doc_frozen_callparams_note,
+    '_doc_random_state': _doc_random_state
+}
+
+
+class normal_invgamma_gen(multi_rv_generic):
+    r"""
+    A normal inverse gamma random variable.
+
+    Methods
+    -------
+    ``pdf(x, sig2, loc=0, variance_scale=1, shape=1, scale=1)``
+        Probability density function.
+    ``logpdf(x, sig2, loc=0, variance_scale=1, shape=1, scale=1)``
+        Log of the probability density function.
+    ``rvs(loc=0, variance_scale=1, shape=1, scale=1, size=1, random_state=None)``
+        Draw random samples from normal and inverse gamma distribution.
+    ``mean(loc=0, variance_scale=1, shape=1, scale=1)``
```

Comment:
The ``mean`` and ``mode`` methods don't seem to do much. Other distributions do have them, so maybe worth keeping just for consistency. The descriptions should be changed though, there are no random variates involved here.